src/CommunityToolkit.Maui.Core/Capabilities/StatusBar/StatusBar.android.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Diagnostics;
+using Android.OS;
+using Android.Util;
+using Android.Views;
+using Microsoft.Maui.Platform;
+using Activity = Android.App.Activity;
+using Debug = System.Diagnostics.Debug;
+
+namespace CommunityToolkit.Maui.Core.Capabilities;
+static partial class StatusBar
+{
+	static void PlatformSetColor(Color color)
+	{
+		if (!IsSupported())
+		{
+			return;
+		}
+		Activity.Window?.SetStatusBarColor(color.ToPlatform());
+	}
+
+	static void PlatformSetStyle(StatusBarStyle style)
+	{
+		if (!IsSupported())
+		{
+			return;
+		}
+
+		switch (style)
+		{
+			case StatusBarStyle.DarkContent:
+				AddBarAppearanceFlag(Activity, (StatusBarVisibility)SystemUiFlags.LightStatusBar);
+
+				break;
+			default:
+				RemoveBarAppearanceFlag(Activity, (StatusBarVisibility)SystemUiFlags.LightStatusBar);
+				break;
+		}
+	}
+
+
+	static void AddBarAppearanceFlag(Activity activity, StatusBarVisibility flag) =>
+			SetBarAppearance(activity, barAppearance => barAppearance |= flag);
+
+
+	static void RemoveBarAppearanceFlag(Activity activity, StatusBarVisibility flag) =>
+		SetBarAppearance(activity, barAppearance => barAppearance &= ~flag);
+
+
+	static void SetBarAppearance(Activity activity, Func<StatusBarVisibility, StatusBarVisibility> updateAppearance)
+	{
+		var window = GetCurrentWindow(activity);
+#pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
`SystemUiVisibility` is deprecated as of API 30 (Android 11). Have you tested this code on Android 11-13 devices?
This worked in XCT since it targeted lover version of Android.

Here're needed changes to support Android 11+: [xamarin/XamarinCommunityToolkit@`38e872a` (#812)](https://github.com/xamarin/XamarinCommunityToolkit/pull/812/commits/38e872a1c54c4956ef0bc62edbb423059eb5d426)

Reply:
I tested it on Android 12 and it worked. We (Core team members) know this is deprecated, but we want to release this feature ASAP, so we will use the obsolete version for now and implement the new approach soon. I suppose that you want to contribute to that, if so, I'll make sure to ping and assign you to this task when it's created.

Reply:
@pictos - could you open an Issue for the use of this deprecated before we merge this PR? That way we won't forget to fix it in the future!

Reply:
@brminnick here's the issue
- https://github.com/CommunityToolkit/Maui/issues/626
@maxkoshevoi if you want to work on that, please add a comment on the issue until Friday.

Reply:
Hm, strange how it works while being deprecated. I'm sure it was crashing on Android 11 device when I retargeted XCT to Android 11, and used deprecated code.
Thank you for creating an issue, added a message there.